### PR TITLE
test: don't check for FIPS provider if it isn't being built

### LINKED ISSUE #19396: OpenSSL 3.0.6 Fails in Rand code on NonStop ia64 (big endian)
The  3.0.6 release has an issue as follows:
```
05-test_pbe.t ...................... ok

    # ERROR: (bool) 'EVP_RAND_generate(z, buf2, sizeof(buf2), 0, 1, NULL, 0) == true' failed @ /home/jenkins/.jenkins/workspace/OpenSSL-3.0_Pipeline/test/drbgtest.c:869
    # false
    # 00000000:error:1C8000BD:Provider routines:ossl_prov_drbg_reseed:error retrieving entropy:/home/jenkins/.jenkins/workspace/OpenSSL-3.0_Pipeline/providers/implementations/rands/drbg.c:584:
    # 00000000:error:1C8000C5:Provider routines:ossl_prov_drbg_generate:reseed error:/home/jenkins/.jenkins/workspace/OpenSSL-3.0_Pipeline/providers/implementations/rands/drbg.c:676:
    # 00000000:error:1C8000BF:Provider routines:ossl_drbg_get_seed:generate error:/home/jenkins/.jenkins/workspace/OpenSSL-3.0_Pipeline/providers/implementations/rands/drbg.c:180:
    # 00000000:error:1C8000BD:Provider routines:ossl_prov_drbg_reseed:error retrieving entropy:/home/jenkins/.jenkins/workspace/OpenSSL-3.0_Pipeline/providers/implementations/rands/drbg.c:584:
    # 00000000:error:1C8000C5:Provider routines:ossl_prov_drbg_generate:reseed error:/home/jenkins/.jenkins/workspace/OpenSSL-3.0_Pipeline/providers/implementations/rands/drbg.c:676:
    # 00000000:error:1C8000BF:Provider routines:ossl_drbg_get_seed:generate error:/home/jenkins/.jenkins/workspace/OpenSSL-3.0_Pipeline/providers/implementations/rands/drbg.c:180:
    # 00000000:error:1C8000BD:Provider routines:ossl_prov_drbg_reseed:error retrieving entropy:/home/jenkins/.jenkins/workspace/OpenSSL-3.0_Pipeline/providers/implementations/rands/drbg.c:584:
    # 00000000:error:1C8000C5:Provider routines:ossl_prov_drbg_generate:reseed error:/home/jenkins/.jenkins/workspace/OpenSSL-3.0_Pipeline/providers/implementations/rands/drbg.c:676:
    # 00000000:error:030000D6:digital envelope routines:evp_rand_generate_locked:generate error:/home/jenkins/.jenkins/workspace/OpenSSL-3.0_Pipeline/crypto/evp/evp_rand.c:560:
    # OPENSSL_TEST_RAND_ORDER=1665521042
    not ok 3 - test_rand_prediction_resistance
```
The Configuration is:
```



Command line (with current working directory = .):

    perl ./Configure nonstop-nse_64 --prefix=/usr/local-ssl3.0 --openssldir=/usr/local-ssl3.0/ssl no-fips --with-rand-seed=egd

Perl information:

    perl
    5.26.3 for NSE-W-nonstop_kernel

Enabled features:

    aria
    async
    autoalginit
    autoerrinit
    autoload-config
    bf
    blake2
    bulk
    cached-fetch
    camellia
    capieng
    cast
    chacha
    cmac
    cmp
    cms
    comp
    ct
    deprecated
    des
    dgram
    dh
    dsa
    dso
    dtls
    dynamic-engine
    ec
    ec2m
    ecdh
    ecdsa
    egd
    engine
    err
    filenames
    gost
    idea
    legacy
    loadereng
    makedepend
    md4
    mdc2
    module
    multiblock
    nextprotoneg
    ocb
    ocsp
    padlockeng
    pic
    pinshared
    poly1305
    posix-io
    psk
    rc2
    rc4
    rdrand
    rfc3779
    rmd160
    scrypt
    secure-memory
    seed
    shared
    siphash
    siv
    sm2
    sm3
    sm4
    sock
    srp
    srtp
    sse2
    ssl
    ssl-trace
    static-engine
    stdio
    tests
    tls
    ts
    ui-console
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method

Disabled features:

    acvp-tests          [cascade]        OPENSSL_NO_ACVP_TESTS
    afalgeng            [not-linux]      OPENSSL_NO_AFALGENG
    asan                [default]        OPENSSL_NO_ASAN
    asm                 [no asm_arch]    OPENSSL_NO_ASM
    buildtest-c++       [default]
    crypto-mdebug       [default]        OPENSSL_NO_CRYPTO_MDEBUG
    devcryptoeng        [default]        OPENSSL_NO_DEVCRYPTOENG
    ec_nistp_64_gcc_128 [default]        OPENSSL_NO_EC_NISTP_64_GCC_128
    external-tests      [default]        OPENSSL_NO_EXTERNAL_TESTS
    fips                [option]
    fips-securitychecks [cascade]        OPENSSL_NO_FIPS_SECURITYCHECKS
    fuzz-afl            [default]        OPENSSL_NO_FUZZ_AFL
    fuzz-libfuzzer      [default]        OPENSSL_NO_FUZZ_LIBFUZZER
    ktls                [default]        OPENSSL_NO_KTLS
    md2                 [default]        OPENSSL_NO_MD2 (skip crypto/md2)
    msan                [default]        OPENSSL_NO_MSAN
    rc5                 [default]        OPENSSL_NO_RC5 (skip crypto/rc5)
    sctp                [default]        OPENSSL_NO_SCTP
    threads             [config]
    trace               [default]        OPENSSL_NO_TRACE
    ubsan               [default]        OPENSSL_NO_UBSAN
    unit-test           [default]        OPENSSL_NO_UNIT_TEST
    uplink              [no uplink_arch] OPENSSL_NO_UPLINK
    weak-ssl-ciphers    [default]        OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                [default]
    zlib-dynamic        [default]
    ssl3                [default]        OPENSSL_NO_SSL3
    ssl3-method         [default]        OPENSSL_NO_SSL3_METHOD

Config target attributes:

    AR => "ar",
    ARFLAGS => "qc",
    CC => "c99",
    HASHBANGPERL => "/usr/bin/env perl",
    RANLIB => "ranlib",
    RC => "windres",
    bn_ops => "SIXTY_FOUR_BIT_LONG RC4_CHAR RC4_CHUNK DES_INT DES_UNROLL BF_PTR",
    build_file => "Makefile",
    build_scheme => [ "unified", "unix" ],
    cflags => "-g -O2 -Wextensions -Wnowarn=203,220,272,734,770,1506 -Wbuild_neutral_library -Wverbose -Wtarget=tns/e -Wsystype=oss -Wlp64 -WIEEE_float",
    cppflags => "",
    defines => [ "OPENSSL_BUILDING_OPENSSL", "OPENSSL_VPROC=\$(OPENSSL_VPROC)", "_XOPEN_SOURCE", "_XOPEN_SOURCE_EXTENDED=1", "_TANDEM_SOURCE", "B_ENDIAN", "_TANDEM_ARCH=2" ],
    disable => [ "threads" ],
    dso_scheme => "DLFCN",
    enable => [ "egd" ],
    ex_libs => "-lrld",
    includes => [  ],
    lflags => "-Weld=\"-set systype oss\" -Weld=\"-set floattype ieee_float\"",
    lib_cflags => "",
    lib_cppflags => "",
    lib_defines => [  ],
    module_cflags => "",
    module_cxxflags => undef,
    module_ldflags => "-Wshared",
    multilib => "64",
    perl => "/usr/bin/perl",
    perl_platform => "Unix",
    shared_argfileflag => "-Weld_obey=",
    shared_cflag => "",
    shared_defflag => "-Weld_obey=",
    shared_defines => [  ],
    shared_extension => ".so",
    shared_ldflag => "-Wshared",
    shared_rcflag => "",
    shared_target => "nonstop-shared",
    sys_id => "TANDEM",
    thread_defines => [  ],
    thread_scheme => "(unknown)",
    unistd => "<unistd.h>",

Recorded environment:

    AR =
    ARFLAGS =
    AS =
    ASFLAGS =
    BUILDFILE =
    CC =
    CFLAGS =
    CPP =
    CPPDEFINES =
    CPPFLAGS =
    CPPINCLUDES =
    CROSS_COMPILE =
    CXX =
    CXXFLAGS =
    HASHBANGPERL =
    LD =
    LDFLAGS =
    LDLIBS =
    MT =
    MTFLAGS =
    OPENSSL_LOCAL_CONFIG_DIR =
    PERL =
    RANLIB =
    RC =
    RCFLAGS =
    RM =
    WINDRES =
    __CNF_CFLAGS =
    __CNF_CPPDEFINES =
    __CNF_CPPFLAGS =
    __CNF_CPPINCLUDES =
    __CNF_CXXFLAGS =
    __CNF_LDFLAGS =
    __CNF_LDLIBS =

Makevars:

    AR              = ar
    ARFLAGS         = qc
    CC              = c99
    CFLAGS          =
    CPPDEFINES      =
    CPPFLAGS        =
    CPPINCLUDES     =
    CXXFLAGS        =
    HASHBANGPERL    = /usr/bin/env perl
    LDFLAGS         =
    LDLIBS          =
    PERL            = perl
    RANLIB          = ranlib
    RC              = windres
    RCFLAGS         =

NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    Makefile

build file:

    Makefile

build file templates:

    Configurations/common0.tmpl
    Configurations/unix-Makefile.tmpl
```

### PR DESCRIPTION
**Pull request by paulidale**
Fixes #19396

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/drbgtest.c [hunk 1]
```diff
@@ -137,6 +137,7 @@ static time_t reseed_time(EVP_RAND_CTX *drbg)
  */
 static int using_fips_rng(void)
 {
+#ifdef FIPS_MODULE
```
Comment (mattcaswell) [inline test/drbgtest.c:140]: This doesn't make sense to me. Isn't this only defined if we are building code which is going to be linked into fips.so? This test will never define it - which means this change results in `using_fips_rng` always returning 0 regardless of whether FIPS is in use or not.

Comment (paulidale) [reply test/drbgtest.c:140]: Hmmmm, other tests do similar and this test did before my earlier change.
I'm not seeing how it gets passed through however.

Never running the test would (which is what happened previously) would make it pass on non-stop.  So mystery solved there.

Comment (mattcaswell) [reply test/drbgtest.c:140]: > other tests do similar

I only see it in test/ecdsatest.c - which also looks like an incorrect usage.

**Comment by paulidale** [comment]
Unrequited and unrequired.

**Pull request closed without merging**